cpp/src/io/comp/nvcomp_adapter.cpp

```diff
@@ -46,11 +46,11 @@
 
 namespace cudf::io::nvcomp {
 
-#if NVCOMP_HAS_TEMPSIZE_EX
 // Dispatcher for nvcompBatched<format>DecompressGetTempSizeEx
 template <typename... Args>
-auto batched_decompress_get_temp_size_ex(compression_type compression, Args&&... args)
+nvcompStatus_t batched_decompress_get_temp_size_ex(compression_type compression, Args&&... args)
```

Comment:
I guess inference fails now because for a given value of `NVCOMP_HAS_TEMPSIZE_EX` (true or false) we can now sometimes return a value _or_ sometimes error?